src/java-test-runner.api.ts

```diff
@@ -0,0 +1,324 @@
+     */
+    Failed = 4,
+
+    /**
+     * Test run has been skipped.
+     */
+    Skipped = 5,
+
+    /**
+     * Test run failed for some other reason (compilation error, timeout, etc).
+     */
+    Errored = 6,
+}
+
+/**
+ * @todo Proposed API
+ * The event that should be emitted when the test run is finished.
+ */
+export interface TestFinishEvent {
+    /**
+     * The status of the test run.
+     */
+    status: number;
+
+    /**
+     * The message of the test run.
+     */
+    message?: string;
+}
+
+/**
+ * @todo Proposed API
+ * The context for running tests.
+ */
+export interface IRunTestContext {
+    /**
+     * The flag to indicate whether the test run is in debug mode.
+     */
+    isDebug: boolean;
+
+    /**
+     * The kind of the test.
+     */
+    kind: TestKind;
+
+    /**
+     * The name of the project.
+     */
+    projectName: string;
+
+    /**
+     * The test items to run.
+     */
+    testItems: vscode.TestItem[];
+
+    /**
+     * VS Code's TestRun object for this test execution.
+     */
+    testRun: vscode.TestRun;
+
+    /**
+     * The workspace folder where the tests are run.
+     */
+    workspaceFolder: vscode.WorkspaceFolder;
+
+    /**
+     * The profile for this test run.
+     */
+    profile?: vscode.TestRunProfile;
+
+    /**
+     * The configuration for this test run.
+     */
+    testConfig?: IExecutionConfig;
+}
+
+/**
+ * @todo Proposed API
+ * The test kind.
+ */
+export enum TestKind {
+    JUnit5 = 0,
+    JUnit = 1,
+    TestNG = 2,
+    None = 100,
+}
+
+/**
+ * @todo Proposed API
+ * The test level.
+ */
+export enum TestLevel {
+    Root = 0,
+    Workspace = 1,
+    WorkspaceFolder = 2,
+    Project = 3,
+    Package = 4,
+    Class = 5,
+    Method = 6,
+    Invocation = 7,
+}
+
+/**
+ * @todo Proposed API
+ * The configuration for running tests.
+ */
+export interface IExecutionConfig {
+    /**
+     * The name of the configuration item.
+     * @since 0.14.0
+     */
+    name?: string;
+
+    /**
+     * The working directory when running the tests.
+     * @since 0.14.0
+     */
+    workingDirectory?: string;
+
+    /**
+     * The classpaths defined in this setting will be appended to the resolved classpaths.
+     * @since 0.33.0
+     */
+    classPaths?: string[]
+
+    /**
+     * The modulepaths defined in this setting will be appended to the resolved modulepaths
+     * @since 0.33.0
+     */
+    modulePaths?: string[]
+
+    /**
+     * The command line arguments which will be passed to the test runner.
+     * @since 0.14.0
+     */
+    args?: any[];
+
+    /**
+     * The path to java executable to use. If undefined project JDK's java executable is used.
+     * @since 0.40.0
+     */
+    javaExec?: string;
+
+    /**
+     * the extra options and system properties for the JVM.
+     * It's deprecated, we should align with the debug launch configuration, which is 'vmArgs'.
+     * @since 0.14.0
+     */
+    vmargs?: any[];
```

Comment:
Remove it. since you're exposing new API, I don't think it's necessary to add deprecated here.